### ISSUE TITLE
Cursor line changes the highlight of folded text. 

### ISSUE THREAD

**Issue by anthony-S93** [olimorris/onedarkpro.nvim#161]
### Issue/Question/Enhancement

I'm not really sure if this is an issue, but for some reason the cursor line doesn't  seem to respect the "Folded" highlight group.

![cursorline](https://user-images.githubusercontent.com/69449791/222050920-ff791b47-4ed3-43fa-8a16-aeaa222cbdba.gif)

**Comment by olimorris** [comment]
This is the desired behaviour unfortunately. If I recall, the reason for this is that when you fold a line, Vim does the calculations to determine what code should be hidden and what the folded line should display. So it doesn't see the fold as `lua` code and as such doesn't know how to apply the syntax highlighting.

You do have the `Folded` highlight group available to you to change the gray foreground.

Or...I highly recommend [nvim-ufo](https://github.com/kevinhwang91/nvim-ufo). As you can see from my fold highlighting below:

<img width="991" alt="Screen Shot 2023-03-01 at 09 33 15@2x" src="https://user-images.githubusercontent.com/9512444/222099783-2f4001a9-1cd8-4a6e-a9fa-a01f4e9aa562.png">

**Comment by olimorris** [comment]
Sorry, I completely misread your question and that it relates to `CursorLine`. What Neovim version are you using? I see that https://github.com/neovim/neovim/pull/22242 applies cursorline colors to folds.

Also, I'd say nvim-ufo could still be the way to go.

**Comment by anthony-S93** [comment]
> What Neovim version are you using?

I'm using `v0.8.3`. 

> You do have the `Folded` highlight group available to you to change the gray foreground.

Indeed, I'm aware of this. The orange color you saw in the gif I shared was the color I set with `vim.api.nvim_set_hl(0, "Folded", {fg = "#FF7518"})`. When I moved the cursor away, it showed the color did change to orange, but not when the cursor line was active. Somehow, the highlight group only works when the cursorline isn't there. 


> Also, I'd say nvim-ufo could still be the way to go.
I saw your screenshots. Looks very promising. I'm going to go ahead and give `nvim-ufo` a try. Thanks for the recommendation.
